### PR TITLE
RUVseq RNA Batch Correction

### PR DESCRIPTION
**Pull request by aadamk**
#### What scientific question is your analysis addressing?

This analysis addresses the issue of apparent batch effects by RNA-library in transcriptome data, correcting specifically for the factors that result in differences across polyA, ribo-deplete, and RNA exome libraries. This analysis should more effectively control for Type I and Type II errors when running differential gene expression analysis by DESeq2. 

#### What was your approach?

This model assumes that differences in housekeeping genes from the [HRT atlas v1.0](https://housekeeping.unicamp.br/) across DESeq2 comparison groups will specifically be a result of technical, rather than biological, variation, whether in a tumor-only or tumor-normal context. Therefore, RUVg uses HRT atlas genes as a negative control gene set for modeling factors of unwanted variation. 

To select the model with the optimal sensitivity/specificity balance, this module requires that users specify a positive control gene set, simply a list of hugo gene names in an rds file containing genes that are expected to differ across comparison groups. A few use cases are specified in the run scripts, namely:

1. TARGET Neuroblastoma: MYC-N amplified versus WT as specified in molecular_subtype from v10 (now deprecated in v11), using `KIM_MYCN_AMPLIFICATION_TARGETS_[UP,DN]` as the positive control gene sets from C2 canonical pathways from MSigDb. 
2. Cancer groups 'High-grade glioma/astrocytoma' and 'Diffuse midline glioma', using cancer stemness genes from https://doi.org/10.1186/s12915-022-01324-0, [Additional file 4: Table S2.](https://static-content.springer.com/esm/art%3A10.1186%2Fs12915-022-01324-0/MediaObjects/12915_2022_1324_MOESM4_ESM.xlsx). `Under Merged_differential_expression tab`, genes whose adj p-value across K27M vs WT day5 < 0.05. 
3. 'High-grade glioma/astrocytoma' and 'Diffuse midline glioma' versus GTEx normals, specifying Reactome Cell Cycle pathway genes as positive control genes (non-redundant gene list from Reactome pathway IDs: R-HSA-69278, R-HSA-69620, and R-HSA-1640170). Note: it is recommended to drop the first two factors of variation (`drop = 2` parameter in RUVg call) to ensure that biological variation across tumor versus normal is maintained (it is expected that biological variation, rather than RNA_library will be the main source of variation). 

#### Which areas should receive a particularly close look?

Performs DESeq2 analysis using [RUVg-based](https://bioconductor.org/packages/devel/bioc/vignettes/RUVSeq/inst/doc/RUVSeq.pdf) batch correction. 

Key steps: 
1. `01-ruvseq-deseq.R` will run tumor-only standard differential expression analysis (no RUVg correction, by molecular_subtype) from DESeq2, to compare to RUVg using k=1-5 factors of variation in a batch-corrected DESeq2 module. 
4. `02-ruvseq-deseq-tn.R` will run tumor-normal standard differential expression analysis (no RUVg correction, by molecular_subtype) from DESeq2, to compare to RUVg using k=1-5 factors of variation in a batch-corrected DESeq2 module. 
5. `03-ruvseq-summarization.R` will select the batch correction method with the optimal balance of sensitivity and specificity, returning normalized counts .RDS file and show the UMAP clustering of the groups of interest. 

#### Is there anything that you want to discuss further?

A neuroblastoma tumor-normal use case may be valuable to test whether certain cancer genes/biomarkers of interest show the expected patterns across tumor versus normal (e.g. a cohort-level box plot). 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

1. Plots: UMAP plots showing clustering of samples before and after batch correction under `plots` directory. P-value histograms showing the impact on the preponderance of significant calls before and after batch correction are also included. 
2. Output: normalized counts of the optimal model are provided in the `output/[dataset]/normalized_counts` folder
3. Output: differential gene expression tabular data are provided for batch corrected and non batch-corrected analyses. Further, RUVseq output are provided for each k should the user desire to compare to optimal output. 

#### What is your summary of the results?

1. From Neuroblastoma comparison, non-batch corrected DGE analysis resulted in a significant nominal p-value for MYCN expression, but not a significant adjusted p-value. Batch correction using 3 factors of variation (the k chosen by this algorithm), resulted in an adjusted p-value of 0.01 for MYCN, which would be expected in amplified versus WT tumors. 
2. From the HGG/DMG comparison, 3, rather than 2, batches were present in the data (RNA exome, poly-A, ribo-deplete), and the UMAP plot suggests that those batch effects are effectively corrected for at k=2. 
3. From HGG/DMG versus normal, the UMAP plot suggests that tumor and normal remain separated even after batch effect correction, though brought in closer proximity (suggesting that biological variation is preserved, but technical variation may be in part corrected for). 

Note: files in `code/archive` subdirectory can be ignored but are useful to re-run matched sample analysis

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
Note: fixes https://github.com/PediatricOpenTargets/ticket-tracker/issues/279

**Comment by jharenza** [comment]
Hi @aadamk when you are ready with this, can you please add this module to the yaml file being run in GA? Thanks!

YAML file here: https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/.github/workflows/run-analysis.yml

**Comment by jharenza** [comment]
Just converting this to draft to not have it mistakenly merged before we make the required updates.

**Review by jharenza — COMMENTED** [review]
I haven't had a chance to look over this in detail, but noticed a few things:
1. Can you add info in the README as to where each input file came from, specifically if there are links to external sources, please provide those for each file. If they were derived, please explain that in more detail in the README. 
2. Can you delete the entire `archive` folder and contents? It is now in git commit history, so we don't need to keep the files around.
3. In `util`, are `ruvg_test.R` and `ruvr_test.R` still "test" scripts? If not being used, can we delete, and if being used, rename to not say "test"? Seems like they are still in use.
4. It would be good to have only one shell script for this module, to conform to the same standards as other modules. 
5. Was this run with docker, and are all required packages in the docker image? 
6. Upon very very quick look, scripts 01 and 02 look to have a good bit of duplicated code - can these be combined into one script with a param or option for running tumor only or tumor vs normal? 

Will make some stylistic suggestions on the code as well.

File: Dockerfile [hunk 1]
```diff
@@ -133,6 +133,7 @@ RUN ./install_bioc.r \
     spatial \
     treemap \
     umap  \
+    uwot \
```
Comment (jharenza) [inline Dockerfile]: Typically changes to the dockerfile should be in a separate PR - can you make a separate PR for this change?

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+}
+
+# source functions
+source('util/deseq2_pvals_histogram.R') # DESeq2 pval histograms
+source('util/edaseq_plot.R') # PCA and UMAP clustering
+source('util/ruvg_test.R') # function to run RUVg
+
+# read histology
+message("Reading in histologies file")
+htl_df <- readr::read_tsv('../../data/v10/histologies.tsv')
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]:
```suggestion
htl_df <- read_tsv(file.path(data_dir, "histologies.tsv"), guess_max = 100000)
```

Can you change all of the paths to use file.path() and use the current data directory (which also should be added above) without direct links to a version?

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+# read expected counts
+message("Reading RSEM expected counts file")
+cnt_df <-
+  readRDS('../../data/v10/gene-counts-rsem-expected_count-collapsed.rds')
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]: Same comment as above - use current data dir, filepath

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+# filter histology
+selected_htl_df <- htl_df %>%
+  dplyr::filter(
+    experimental_strategy == "RNA-Seq",
+    sample_type == "Tumor",
+    cancer_group %in% cancer_group_values,!is.na(molecular_subtype)
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]:
```suggestion
    cancer_group %in% cancer_group_values,
    !is.na(molecular_subtype)
```

just breaking this up for readability

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+  edgeR::filterByExpr(counts_object, group = molecular_subtype)
+counts_object <-
+  counts_object[counts_object_filtered, , keep.lib.sizes = FALSE]
+
+# create new expression set
+seq_expr_set <-
+  EDASeq::newSeqExpressionSet(
+    counts = round(counts_object$counts),
+    phenoData = data.frame(bs_id, molecular_subtype, RNA_library, row.names = colnames(round(
+      counts_object$counts
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R:136]:
```suggestion
    phenoData = data.frame(bs_id, factor(as.character(selected_htl_df$molecular_subtype)), factor(as.character(selected_htl_df$RNA_library)), row.names = colnames(round(
      counts_object$counts
```
Would this work? Then you can remove lines 108 & 109. Would also have to change in other places.

File: analyses/rnaseq-batch-correct/code/02-ruvseq-deseq-tn.R [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+  width = 8,
+  height = 7,
+  bg = "white"
+)
+
+print("DESeq2 analysis finished.")
+
+# Read in positive and negative control gene sets
+pos_ctrl_genes <- readRDS(file.path(input_dir, pos_ctrl_genes.f))
+gene.list <- c(neg_ctrl_genes, pos_ctrl_genes)
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/02-ruvseq-deseq-tn.R]:
```suggestion
pos_neg_ctrl_genes <- c(neg_ctrl_genes, pos_ctrl_genes)
```
Suggest to make this more informative

File: analyses/rnaseq-batch-correct/code/02-ruvseq-deseq-tn.R [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+  height = 7,
+  bg = "white"
+)
+
+
+# Return normalized counts
+norm_count_dir <- file.path(output_dir, 'normalized_counts')
+if (!dir.exists(norm_count_dir)) {
+  dir.create(norm_count_dir, recursive = TRUE)
+}
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/02-ruvseq-deseq-tn.R]: Can you move this up to the directory creation section at the top please?

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+  width = 8,
+  height = 7,
+  bg = "white"
+)
+
+# Return normalized counts
+norm_count_dir <- file.path(output_dir, 'normalized_counts')
+if (!dir.exists(norm_count_dir)) {
+  dir.create(norm_count_dir, recursive = TRUE)
+}
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]: Can you move this to the top of the script with other directories please?

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+  height = 7,
+  bg = "white"
+)
+
+
+print("DESeq2 analysis finished.")
+
+# Read in positive and negative control gene sets
+pos_ctrl_genes <- readRDS(file.path(input_dir, pos_ctrl_genes.f))
+gene.list <- c(neg_ctrl_genes, pos_ctrl_genes)
```
Comment (jharenza) [inline analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]: Can probably also add these at the top of the script before analyses, so all file inputs are grouped in a section together

**Review by jharenza — CHANGES_REQUESTED** [review]
Thanks for this extensive work @komalsrathi and @aadamk. I made a few stylistic comments and also suggest cohesive variable naming throughout, if possible where there is a back and forth of periods and underscores used (for ex - `output_dir` vs `output.dir`). Generally, the scripts are easy to follow, but had a suggestion to combine into one R script and one shell script so code is not so redundant - let me know your thoughts!

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R (hunk 2 above)
Comment (aadamk) [reply analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]: yes. 
currently, it is written to compare tumor-only based on molecular subtype. since MYCN subtypes were migrated to pathology free text for NBL, I hardcoded temporarily for v10 but will change as described above once end-to-end testing by reviewers is complete.

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R (hunk 3 above)
Comment (aadamk) [reply analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R]: will do the same here

**Comment by aadamk** [comment]
> Thanks for this extensive work @komalsrathi and @aadamk. I made a few stylistic comments and also suggest cohesive variable naming throughout, if possible where there is a back and forth of periods and underscores used (for ex - `output_dir` vs `output.dir`). Generally, the scripts are easy to follow, but had a suggestion to combine into one R script and one shell script so code is not so redundant - let me know your thoughts!

Thank you for the review @jharenza 
I have changed to one shell script and will address the minor changes. 

Regarding changing to one R script - I was considering this but kept them divided for a few reasons. One, T-N uniquely creates a newly coded deseq2 group based on whether the specimen is derived tumor versus normal tissue, whereas tumor-only uses molecular_subtype directly as a DESeq2 group. Second, T-N uniquely requires that the user specifies gtex subgroups. Third, while both have a 'drop' parameter, T-N is likely to require a default of '2' for a successful batch correction, whereas this is less likely to be used for tumor-only. 

So in summary, since T-N is creating groups in a distinct way and the underlying algorithm is run a bit differently, I was thinking that 'compartmentalizing' the two comparison types may be clearer for any user of the module despite the shared code (as combining would require some conditionals and additional lines of code in an already long script). Open to other perspectives, though.

File: analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R (hunk 5 above)
Comment (aadamk) [reply analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R:136]: i hesitate to change given that this is used in other parts of the script, namely to build the deseq2 dataset as well as the seq expression set for non-batch DGE analysis and plotting, respectively.

File: Dockerfile.txt [hunk 10]
```diff
@@ -412,7 +412,8 @@ RUN ./install_bioc.r \
 
 # Package for removing unwanted variation from RNA-Seq data
 RUN ./install_bioc.r \
-    RUVSeq
```
Comment (jharenza) [inline Dockerfile.txt]: I think this is why you have a merge conflict - can just carry over Dockerfile from dev, as this one has `.txt`, which it should not

Comment (jharenza) [reply Dockerfile.txt]: Oh I see - you did pull over, but can you remove the `.txt` extension?

**Comment by aadamk** [comment]
all major changes have been made and both tumor-only and T/N are confirmed to run within the docker container. I commented out the NBL analysis as the script is designed to compare by molecular_subtype, and v11 NBL subtypes were moved over to pathology free text. If this passes the CI checks, I think this should be ready to merge. 

Note: the dockerfile still shows whitespace differences for reasons I don't understand. Further, I think a future PR can remove [rna-seq-protocol-ruvseq](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/rna-seq-protocol-ruvseq), as this consists of only benchmarking analyses run by Yuanchao in the past.

**Comment by afarrel** [comment]
The corrected count matrices and DGE results for HGG-DMG and NBL genes of interest look as expected plus there are very few major changes with other PMTL genes. The results are suitable for DGE.

**Comment by jharenza** [comment]
@aadamk it looks like the latest error is a file subset error - can you check this one out?

**Comment by aadamk** [comment]
> @aadamk it looks like the latest error is a file subset error - can you check this one out?

Unsure why this is giving an error. This previously ran successfully in the docker environment and is currently running successfully locally as well. 

When checking the sample content of the rnaseq rsem expression file from my v11 data download, specifically the few bs_ids that were listed as not found in the CI error, I did indeed find those ids as columns in the counts file. 
I removed the dplyr::select clause as certain dplyr functions were giving me issues in the past. I instead switched to base R style subsetting, by finding the indices of the columns matching the bs_ids of interest and subsetting the expression dataframe using brackets, but it still appears that certain bs_ids are not being found given the nature of the error downstream (the subsetted histology ids do not intersect perfectly with the ids in the counts dataframe after subsetting). 

Unsure how to proceed - it seems like the data that are downloaded by CI do not match the data that downloaded locally to my machine.

**Comment by ewafula** [comment]
> > @aadamk it looks like the latest error is a file subset error - can you check this one out?
> 
> Unsure why this is giving an error. This previously ran successfully in the docker environment and is currently running successfully locally as well.
> 
> When checking the sample content of the rnaseq rsem expression file from my v11 data download, specifically the few bs_ids that were listed as not found in the CI error, I did indeed find those ids as columns in the counts file. I removed the dplyr::select clause as certain dplyr functions were giving me issues in the past. I instead switched to base R style subsetting, by finding the indices of the columns matching the bs_ids of interest and subsetting the expression dataframe using brackets, but it still appears that certain bs_ids are not being found given the nature of the error downstream (the subsetted histology ids do not intersect perfectly with the ids in the counts dataframe after subsetting).
> 
> Unsure how to proceed - it seems like the data that are downloaded by CI do not match the data that downloaded locally to my machine.

@aadamk, the test data used by the CI is a subset of the v11 expression matrices and can be downloaded using the following  script:  
- https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/scripts/download-testing-files.sh

We might need to create a specific subset for the module if the current CI testing matrix is not suitable.  The module that creates this CI testing subset data sets is here: 
- https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/create-subset-files

Cc @jharenza

**Comment by ewafula** [comment]
@aadamk, here is the ticket for developing the corresponding CAVATICA app by the D3b bix-dev team. You can expound on specific details if any. 
- https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/scripts/download-testing-files.sh

**Comment by aadamk** [comment]
> We might need to create a specific subset for the module if the current CI testing matrix is not suitable. The module that creates this CI testing subset data sets is here:
> 
> * https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/create-subset-files

Thank you for the clarification @ewafula 
Yes, I think that would be helpful to allow this module to pass CI. I can create a ticket for this.

**Comment by ewafula** [comment]
@aadamk, 

@jharenza  suggested the following two options to resolve the issue of the CI failing because of missing bs_ids in the CI testing counts subset matrix:
1) Only run the tumor-normal script when with CI testing. It will require including an environmental variable `OPENPBTA_TESTING` in the `run_ruvseq.sh` bash script to conditionally run the tumor-normal batch correction script and not ignore the tumor-only batch correction script. 
2) Enrich the CI testing counts data matrix with a few bs_ids that will enable the batch correction module to run successfully without failing GA checks. 

To me, it seems both options, including the tumor-normal, require enriching the CI testing counts matrix dataset with as follows: 
1) include a few bs_ids with contrasting molecular subtypes in the CI subset counts matrix for  the `target_nbl` and` hgg_dmg` tumor-only batch correction commands in the bash script (`run_ruvseq.sh`)
2) include corresponding  cancer_group/gtex_subgroup bs_ids in the CI subset counts matrix for the `HGG-DMG_TN`  tumor-normal batch correction command in the bash script (`run_ruvseq.sh`)

Here is an example of [a list of bs_ids](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/4098cbe419bf51d0f88f71a2e4fa5297b79a8832/analyses/create-subset-files/01-get_biospecimen_identifiers.R#L313-L327) included in the CI testing dataset to enrich for TP53 and NF1 mutations determined by [this script](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/create-subset-files/00-enrich-positive-examples.Rmd) in `create-subset-files` module. 

Since the batch correction analysis is upstream, the `histologies.tsv` utilized in the module needs to be replaced with `histologies-base.tsv`. It might require temporarily creating the subtype group (i.e., MYCN status via `pathology_free_text_diagnosis`) and changing to use `broad_histoiogy` or `short_histology` and not cancer group in the scripts

@aadamk,  would you mind adding more detail, specifically for task 1 in [ticket #460](https://github.com/PediatricOpenTargets/ticket-tracker/issues/460) with requirements on how to select bs_ids that would enrich the CI testing counts matrix for the batch correction module? It will be helpful when updating the [create-subset-files](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/create-subset-files) module.

**Comment by jharenza** [comment]
Hi @aadamk! This passed with the latest updates @devbyaccident did over in #295 and I have now merged #297. Just a few more requests before I approve:

1. Can you bring the dev version of the dockerfile into your PR (there are some spacing changes which are changing the dockerfile here). Not a huge deal since the docker image will build anyway, but better to keep any dockerfile changes out of this since it is not needed.
2. Can you update this module in the analysis [README](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/README.md), as well as links/pertinent info in the older modules so all relevant batch correction modules are up to date? Eg `rna-seq-protocol-ruvseq` and `rna-seq-protocol-dge` have broken links, the latter says in progress. Related, do we need to keep those modules around or does your new module take care of everything from those previous modules?


So close! :)

**Comment by aadamk** [comment]
> Hi @aadamk! This passed with the latest updates @devbyaccident did over in #295 and I have now merged #297. Just a few more requests before I approve:
> 
> 1. Can you bring the dev version of the dockerfile into your PR (there are some spacing changes which are changing the dockerfile here). Not a huge deal since the docker image will build anyway, but better to keep any dockerfile changes out of this since it is not needed.
> 2. Can you update this module in the analysis [README](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/README.md), as well as links/pertinent info in the older modules so all relevant batch correction modules are up to date? Eg `rna-seq-protocol-ruvseq` and `rna-seq-protocol-dge` have broken links, the latter says in progress. Related, do we need to keep those modules around or does your new module take care of everything from those previous modules?
> 
> So close! :)

Hi @jharenza 
These are now addressed
For 1, the dockerfile contents from `dev` have been copied into the branch dockerfile. 
For 2, I have updated the readme accordingly. `rna-seq-protocol-ruvseq` and `rna-seq-protocol-dge` can be removed or archived as these were pilot analyses rather than generalized DGE utilities.

**Comment by jharenza** [comment]
> > Hi @aadamk! This passed with the latest updates @devbyaccident did over in #295 and I have now merged #297. Just a few more requests before I approve:
> > 
> > 1. Can you bring the dev version of the dockerfile into your PR (there are some spacing changes which are changing the dockerfile here). Not a huge deal since the docker image will build anyway, but better to keep any dockerfile changes out of this since it is not needed.
> > 2. Can you update this module in the analysis [README](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/README.md), as well as links/pertinent info in the older modules so all relevant batch correction modules are up to date? Eg `rna-seq-protocol-ruvseq` and `rna-seq-protocol-dge` have broken links, the latter says in progress. Related, do we need to keep those modules around or does your new module take care of everything from those previous modules?
> > 
> > So close! :)
> 
> Hi @jharenza These are now addressed For 1, the dockerfile contents from `dev` have been copied into the branch dockerfile. For 2, I have updated the readme accordingly. `rna-seq-protocol-ruvseq` and `rna-seq-protocol-dge` can be removed or archived as these were pilot analyses rather than generalized DGE utilities.

thanks @aadamk - made a [ticket](https://github.com/PediatricOpenTargets/ticket-tracker/issues/483) for removal of those

**Review by jharenza — APPROVED** [review]

**Comment by aadamk** [comment]
> The corrected count matrices and DGE results for HGG-DMG and NBL genes of interest look as expected plus there are very few major changes with other PMTL genes. The results are suitable for DGE.

Hi @afarrel - thank you for this review. can you approve so that we can merge this in?

**Comment by jharenza** [comment]
We can merge this next - we only need one approving review. Cc @ewafula
